controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,253 @@
+			return reconcile.Result{}, fmt.Errorf("failed to delete key %v in odf-info %v: %v", obj.getOdfInfoKeyName(storageCluster), odfInfoConfigMap.Name, err)
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := OdfDeploymentTypeExternal
+	isDROptimized := false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = OdfDeploymentTypeInternal
+		if isDROptimized, err = getIsDROptimized(r, storageCluster); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+	storageClusterCount := len(r.clusters.GetStorageClusters())
+
+	data := OdfInfoData{
+		OdfVersion:          odfVersion,
+		OdfDeploymentType:   odfDeploymentType,
+		StorageClusterCount: storageClusterCount,
+		IsDROptimized:       isDROptimized,
+		StorageSystemName:   storageSystemName,
+		Clients:             []ConnectedClient{},
+		StorageCluster: InfoStorageCluster{
+			NamespacedName:          types.NamespacedName{Name: storageCluster.Name, Namespace: storageCluster.Namespace},
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getStorageSystemName(storageCluster *ocsv1.StorageCluster) (string, error) {
+	storageSystemRef := util.Find(storageCluster.OwnerReferences, func(ref *metav1.OwnerReference) bool {
+		return ref.Kind == "StorageSystem"
+	})
+	if storageSystemRef != nil {
+		return storageSystemRef.Name, nil
+	}
+
+	return "", fmt.Errorf(
+		"failed to find parent StorageSystem's name in StorageCluster %q"+
+			" ownerreferences, %v",
+		storageCluster.Name,
+		storageCluster.OwnerReferences)
+
+}
+
+func getIsDROptimized(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (bool, error) {
+	var cephCluster rookCephv1.CephCluster
+	err := r.Client.Get(
+		r.ctx,
+		types.NamespacedName{
+			Name:      generateNameForCephClusterFromString(storageCluster.Name),
+			Namespace: storageCluster.Namespace},
+		&cephCluster)
```

Comment:
```suggestion
	var cephCluster rookCephv1.CephCluster
	cephCluster.Name = generateNameForCephClusterFromString(storageCluster.Name)
	cephCluster.Namespace = storageCluster.Namespace
	err := r.Client.Get(
		r.ctx,
		client.ObjectKeyFromObject(cephCluster),
		&cephCluster,
	)
```

Please note the `,` after `&cephCluster` and the move of the last `)` into a separate line